airstable.rb

```diff
@@ -0,0 +1,256 @@
+require 'sinatra'
+require './setup'
+require './models'
+require 'PP'
+
+set(:sessions, true)
+set(:session_secret, ENV['SESSION_SECRET'])
+
+# set(:requires_login) do
+#   condition do
+#     redirect '/' unless user_signed_in?
+#   end
+#   true
+# end
+
+# Before filter that redirects user to login page unless user is logged in
+# or page does not require login
+
+### This eliminates the possibility of a 404 page. Better way to redirect only
+### for valid routes?
+before do
+  request_path = request.path_info.split('/')
+
+  if (['sessions', nil].include? request_path[1]) ||
+     (request_path[1] == 'users' && request_path[2] == 'new')
+    pass
+  else
+    authorize!
+  end
+end
+
+get '/' do
+  user = User.new
+  erb :index, locals: { user: user, title: 'Home' }
+end
+
+get '/sessions/sign_out' do
+  sign_out!
+  redirect('/')
+end
+
+post '/sessions' do
+  user = User.find_by_credentials(params[:email], params[:password])
+
+  if user.errors.any?
+    erb :index, locals: { user: user, title: 'Home' }
+  else
+    sign_in!(user)
+    redirect('/dashboard')
+  end
+end
+
+get '/users/new' do
+  user = User.new
+  erb :'/users/new', locals: { user: user, title: 'Create an Account' }
+end
+
+post '/users/new' do
+  user = User.create(params[:user])
+
+  if user.saved?
+    sign_in!(user)
+    redirect('/dashboard')
+  else
+    erb :'/users/new', locals: { user: user, title: 'Create an Account' }
+  end
+end
+
+get '/users/:id/edit' do
```

Comment:
Ok, was treating this as the "final version," but guessing it's not since you left in these empty routes.